app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -158,11 +158,8 @@ def encryption_opts
     if query_params[:skip_encryption].present? && current_service_provider&.skip_encryption_allowed
       nil
     elsif current_service_provider&.encrypt_responses?
-      cert = saml_request.service_provider.matching_cert ||
-             current_service_provider&.ssl_certs&.first
```

Comment:
the `.first` was masking an issue where we couldn't find a matching cert, so we sent back a wrong cert... this may cause some nil errors in the specs, will see how it goes

Reply:
good news, removing this didn't break the build so it didn't cause a regression that we know about